apps/examples/testcase/le_tc/compression/tc_compress_read.c

```diff
@@ -305,6 +305,46 @@ static void tc_comp_name(void)
 	TC_SUCCESS_RESULT();
 }
 
+/****************************************************************************
+ * Name: tc_comp_file_decomp
+ *
+ * Description:
+ *   Testing decompression of the compressed file
+ *
+ * Returned Value:
+ *   None
+ ****************************************************************************/
+static void tc_comp_file_decomp(void)
+{
+	int ret;
+	fd = get_compfd();
+
+	/*If FAIL : Failed to open testcase driver*/
+	TC_ASSERT_GEQ("get_compfd", fd, OK);
+
+	// Before running check compressed file should exist /mnt/dummy_comp
+	const char *file_path = "/mnt/dummy_comp.dat";
+
+	// initialising decompression framework in kernel
+	ret = ioctl(fd, COMPIOC_FDCMP_INIT, file_path);
+	TC_ASSERT_EQ("compress_init", ret, OK);
+
+	// to get buffer size require to hold decompressed data
+	ret = ioctl(fd, COMPIOC_FDCMP_GET_BUFSIZE, NULL);
+	TC_ASSERT_GT("get_compression_header", ret, OK);
+	uint8_t *dst_buffer = (uint8_t *)malloc(ret);
+
+	// Test decompression method
+	ret = ioctl(fd, COMPIOC_FDCMP_DECOMPRESS, dst_buffer);
+	TC_ASSERT_EQ_CLEANUP("compress_read", ret, OK, free(dst_buffer));
+
+	ret = ioctl(fd, COMPIOC_FDCMP_DEINIT, NULL);
```

Comment:
This looks not thread safe. Is it?

Reply:
Hi Chang.. 
We are storing all the data related compression in fd->fpriv.
So, as long as each thread opens the driver and makes ioctl calls, then it is safe. 
But if a open driver fd is shared accross threads, then it is not safe. It is application's responsibility to prevent this.